compliance/collection/nodeinventorizer/nodeinventory.go

```diff
@@ -58,23 +58,42 @@ func protoComponentsFromScanComponents(c *nodes.Components) *scannerV1.Component
 	return protoComponents
 }
 
-func convertRHELComponents(rc *database.RHELv2Components) []*scannerV1.RHELComponent {
+func convertAndDedupRHELComponents(rc *database.RHELv2Components) []*scannerV1.RHELComponent {
 	if rc == nil || rc.Packages == nil {
 		log.Warn("No RHEL packages found in scan result")
 		return nil
 	}
-	v1rhelc := make([]*scannerV1.RHELComponent, 0, len(rc.Packages))
-	for _, rhelc := range rc.Packages {
-		v1rhelc = append(v1rhelc, &scannerV1.RHELComponent{
-			// TODO(ROX-13936): Find out if ID field is needed here
+
+	convertedComponents := make(map[string]*scannerV1.RHELComponent, 0)
+	for i, rhelc := range rc.Packages {
+		comp := &scannerV1.RHELComponent{
```

Comment:
nit: maybe make this after we ensure we need it so we can reduce allocations/de-allocations when a duplicate is found

Reply:
I did the conversion first with two thoughts in mind:
1. My understanding is that we only expect very rare cases of duplicates, if any at all
2. I wanted to derive the key from the converted object to improve code readability

As with the other nit: Would you be okay with revisiting this when we move the conversion into Scanner?